content/posts/unpublished/migration-from-karma-jasmine-to-jest/post.md

```diff
@@ -0,0 +1,139 @@
+# Migration from Karma/Jasmine to Jest
+
+Given the faster execution, clearer stack trace of Jest, Karma is being deprecated; decided to migrate to Jest.
+
+![](assets/migration_karma_jest.png)
+
+Authors: Utku, Omar Rojas
+Date: unpublished
+Category: frontend
+
+tags: web,testing,test,unit,frontend,karma,jest,jasmine
+
+---
+
+## Introduction
+
+In Angular 16, Karma was deprecated while Jest was introduced in experimental mode. The Angular team moved towards Jest as the recommended testing framework for Angular applications. While Karma was the default testing framework for Angular in the past, Jest is being explored as a potential replacement due to its simplicity, speed, and built-in features like snapshot testing and mocking. However, it’s important to note that Jest is still in the experimental stage in Angular 16 and Angular 17, and using it in production may have limitations or unsupported features.
+
+Given the faster execution, clearer stack trace of Jest, Karma is being deprecated; decided to migrate to Jest.
+
+## Migration
+
+Below steps have been followed for complete migration in retail(universal) web app:
```

Comment:
I think it is better to rewrite this statement because the readers are not necessarily going to know what a retail universal web app is. You can provide a setup statement which tells the reader which kind of app are you applying the below changes to. For eg, "In Backbase, we use Nx angular schematics to generate our applications. Below steps are following that kind of project structure. But if you (the reader) has a different project structure, the steps may still provide you with the direction you want to take.

Reply:
Good catch, thanks